### PR TITLE
Robustify

### PR DESCRIPTION
**Pull request by vincent-miller**
Added current cutoff. Not sure if this is the right way to do it (maybe there is an automatic way to do this in CAN talon)

### CODE REVIEW

File: QuasarHelios2015/src/org/team1540/quasarhelios/Elevator.java [hunk 1]
```diff
@@ -105,8 +105,13 @@ public static void setup() {
...
         };
         
-        FloatMixing.pumpWhen(QuasarHelios.constantControl, Mixing.select((BooleanInputPoll) overrideEnabled, main, override), winch);
-
+        BooleanInput maxCurrent = BooleanMixing.createDispatch(FloatMixing.floatIsAtLeast(winchCAN.asStatus(ExtendedMotor.StatusType.OUTPUT_CURRENT), 
+                ControlInterface.mainTuning.getFloat("elevator-max-current", 40)), QuasarHelios.globalControl);
+        
+        BooleanMixing.onPress(maxCurrent).send(BooleanMixing.getSetEvent(BooleanMixing.combine(raising, lowering), false));
+        
+        FloatMixing.pumpWhen(QuasarHelios.constantControl, Mixing.select(overrideEnabled, main, override), winch);
```
Comment (celskeggs) [inline QuasarHelios2015/src/org/team1540/quasarhelios/Elevator.java]: The `(BooleanInputPoll)` was there to make sure that `select`'s output will always update - not just when `overrideEnabled` changes.

File: QuasarHelios2015/src/org/team1540/quasarhelios/Elevator.java [hunk 2]
```diff
@@ -105,8 +105,13 @@ public static void setup() {
             return f;
         };
         
-        FloatMixing.pumpWhen(QuasarHelios.constantControl, Mixing.select((BooleanInputPoll) overrideEnabled, main, override), winch);
-
+        BooleanInput maxCurrent = BooleanMixing.createDispatch(FloatMixing.floatIsAtLeast(winchCAN.asStatus(ExtendedMotor.StatusType.OUTPUT_CURRENT), 
+                ControlInterface.mainTuning.getFloat("elevator-max-current", 40)), QuasarHelios.globalControl);
+        
+        BooleanMixing.onPress(maxCurrent).send(BooleanMixing.getSetEvent(BooleanMixing.combine(raising, lowering), false));
```
Comment (celskeggs) [inline QuasarHelios2015/src/org/team1540/quasarhelios/Elevator.java]: Might look cleaner by using `setWhen`.

File: QuasarHelios2015/src/org/team1540/quasarhelios/Elevator.java [hunk 3]
```diff
@@ -105,8 +105,13 @@ public static void setup() {
             return f;
         };
         
-        FloatMixing.pumpWhen(QuasarHelios.constantControl, Mixing.select((BooleanInputPoll) overrideEnabled, main, override), winch);
-
+        BooleanInput maxCurrent = BooleanMixing.createDispatch(FloatMixing.floatIsAtLeast(winchCAN.asStatus(ExtendedMotor.StatusType.OUTPUT_CURRENT), 
+                ControlInterface.mainTuning.getFloat("elevator-max-current", 40)), QuasarHelios.globalControl);
```
Comment (celskeggs) [inline QuasarHelios2015/src/org/team1540/quasarhelios/Elevator.java]: Maybe include the units? `elevator-max-current-amps`?

File: QuasarHelios2015/src/org/team1540/quasarhelios/Elevator.java [hunk 4]
```diff
@@ -114,7 +119,9 @@ public static void setup() {
         timer.schedule(elevatorTimeout, BooleanMixing.getSetEvent(BooleanMixing.combine(raising, lowering), false));
         
         BooleanMixing.xorBooleans(raising, lowering).send(timer.getRunningControl());
-
+        
+        
+
```
Comment (celskeggs) [inline QuasarHelios2015/src/org/team1540/quasarhelios/Elevator.java]: ?

**Comment by celskeggs** [comment]
That's what I see right now.

File: QuasarHelios2015/src/org/team1540/quasarhelios/Elevator.java (hunk 2 above)
Comment (vincent-miller) [inline QuasarHelios2015/src/org/team1540/quasarhelios/Elevator.java]: BooleanOutput doesn't have a setWhen method, those are in BooleanStatus

**Comment by celskeggs** [comment]
Looks good. Try it in the emulator. If it works, let me know and I'll merge it.

**Comment by celskeggs** [comment]
Looks good.

**Pull request merged**